src/Compilers/CSharp/Test/Semantic/Semantics/PropertyFieldKeywordTests.cs

```diff
@@ -2731,6 +3378,54 @@ public S(int arg)
             Assert.Equal(0, accessorBindingData.NumberOfPerformedAccessorBinding);
         }
 
+        [Theory, CombinatorialData]
+        public void InStruct_Cycle(bool firstIsSemi, bool secondIsSemi)
+        {
+            var firstAccessor = firstIsSemi ? "get => field;" : "get;";
+            var secondAccessor = secondIsSemi ? "get => field;" : "get;";
+            var comp = CreateCompilation($@"
+struct S1
+{{
+    public S2 P {{ {firstAccessor} }}
+}}
+
+struct S2
+{{
+    public S1 P {{ {secondAccessor} }}
+}}
+");
+            var accessorBindingData = new SourcePropertySymbolBase.AccessorBindingData();
+            comp.TestOnlyCompilationData = accessorBindingData;
+            comp.VerifyDiagnostics(
+                // (4,15): error CS0523: Struct member 'S1.P' of type 'S2' causes a cycle in the struct layout
+                //     public S2 P { get; }
+                Diagnostic(ErrorCode.ERR_StructLayoutCycle, "P").WithArguments("S1.P", "S2").WithLocation(4, 15),
+                // (9,15): error CS0523: Struct member 'S2.P' of type 'S1' causes a cycle in the struct layout
+                //     public S1 P { get; }
+                Diagnostic(ErrorCode.ERR_StructLayoutCycle, "P").WithArguments("S2.P", "S1").WithLocation(9, 15)
+                );
+            Assert.Equal(0, accessorBindingData.NumberOfPerformedAccessorBinding);
+        }
+
+        [Fact]
+        public void InStruct_SelfCycle()
```

Comment:
>InStruct_SelfCycle [](http://example.com/codeflow?start=20&length=18)

I think we should also test if definite assignment analysis for structures like that is going to cause some problems.

Reply:
If this is done by adding a constructor to the struct under test, then it's fixed now.